packages/@aws-cdk/core/lib/cfn-fn.ts

```diff
@@ -356,6 +365,19 @@ class FnFindInMap extends FnBase {
   }
 }
 
+/**
+ * The intrinsic function ``Fn::Transform`` specifies a macro to perform custom processing on part of a stack template.
+ */
+class FnTransform extends FnBase {
+  /**
+   * 
+   * @param macroNameParameters The name of the macro to be invoked, and the parameters to pass to it
+   */
+  constructor(macroNameParameters: string) {
+    super('Fn::Transform', macroNameParameters);
```

Comment:
This isn't correct, right?

[`Fn::Transform`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-transform.html) takes an object with 2 keys: the first one is `"Name"`, and the second is `"Parameters"`, which is a `{ [key: string]: any }` (a mapping from string keys to values).

Reply:
The macroNameParameters parameter contains both the macro name and the parameters for that macro.

Reply:
Right. What I'm saying is that they actually should be modeled separately :). The same way we don't model `Fn::Equals` as a `constructor(leftAndRightSide: string)`, we shouldn't be doing it for `Fn::Transform` either.